src/components/AttachmentCarousel/index.js

```diff
@@ -0,0 +1,204 @@
+
+    componentDidMount() {
+        this.makeAndSetArrowState();
+    }
+
+    componentDidUpdate(prevProps) {
+        const prevReportSize = _.size(prevProps.reportActions);
+        const currReportSize = _.size(this.props.reportActions);
+        if (prevReportSize === currReportSize) {
+            return;
+        }
+        this.makeAndSetArrowState();
+    }
+
+    /**
+     * Toggles the visibility of the arrows
+     * @param {Boolean} showArrows
+     */
+    onShowArrow(showArrows) {
+        this.setState({showArrows});
+    }
+
+    /**
+     * Helps to navigate between next/previous attachments
+     * @param {Object} attachmentItem
+     * @returns {Object}
+     */
+    getAttachment(attachmentItem) {
+        const sourceURL = _.get(attachmentItem, 'sourceURL', '');
+        const file = _.get(attachmentItem, 'file', {name: ''});
+        return {
+            sourceURL: addEncryptedAuthTokenToURL(sourceURL),
+            file,
+        };
+    }
+
+    /**
+     * this will be called to set up an attachment array and finding an index
+     */
+    makeAndSetArrowState() {
+        let page;
+        const actionsArr = _.values(this.props.reportActions);
+        const attachments = _.reduce(actionsArr, (attachmentsAccumulator, reportAction) => {
+            if (reportAction.originalMessage && reportAction.originalMessage.html) {
+                const matchesIt = reportAction.originalMessage.html.matchAll(CONST.REGEX.ATTACHMENT_DATA);
+                const matches = [...matchesIt];
+                if (matches.length === 2) {
+                    const [src, name] = matches;
+                    if ((this.state.sourceURL && src[2].includes(this.state.sourceURL))
+                        || (!this.state.sourceURL && src[2].includes(this.props.sourceURL))) {
+                        page = attachmentsAccumulator.length;
+                    }
+                    const url = src[2].replace(
+                        CONFIG.EXPENSIFY.EXPENSIFY_URL,
+                        CONFIG.EXPENSIFY.URL_API_ROOT,
+                    );
+                    attachmentsAccumulator.push({sourceURL: url, file: {name: name[2]}});
+                }
+            }
+            return attachmentsAccumulator;
+        }, []);
+
+        const {sourceURL, file} = this.getAttachment(attachments[page]);
+        this.setState({
+            page,
+            attachments,
+            sourceURL,
+            file,
+            isBackDisabled: page === 0,
+            isForwardDisabled: page === attachments.length - 1,
+        });
+    }
+
+    /**
+     * increments or decrements the index to get another selected item
+     * @param {Number} deltaSlide
+    */
+    cycleThroughAttachments(deltaSlide) {
+        if ((deltaSlide < 0 && this.state.isBackDisabled) || (deltaSlide > 0 && this.state.isForwardDisabled)) {
+            return;
+        }
+        this.setState(({attachments, page}) => {
+            const nextIndex = page + deltaSlide;
```

Comment:
Are we performing cycling through attachments from `reportComments` which are already fetched right?
What can happen at the end of the cycle, we wouldn't fetch earlier comments attachments, right?

We should trigger fetch and check if are there any attachments received and cycle through it

Reply:
I think I have this correct, but it's tough to test on my own. Do you have a way I could get a bunch of attachments in a chat?